aten/src/ATen/native/native_functions.yaml

```diff
@@ -14736,3 +14736,10 @@
   dispatch:
     CUDA: _fused_adam_kernel_cuda_
   autogen: _fused_adam, _fused_adam.out
+
+# Collectives
+# TODO: add reduce_op and add some form of ranks instead of processgroup obj
+- func: all_reduce(Tensor self, int group_id, str reduce_op) -> Tensor
```

Comment:
quick question: would this `group_id` be just a single process group id across all ranks? for example we have two pgs, and we do all_reduce on two pgs together, would this `group_id` be different across two pgs? If this is the case, I feel this make it not SPMD and I'm wondering we should make this SPMD instead.

Reply:
don't look at this part too closely. we've moved on to a new API proposal with list[rank] but I didn't bother to update this stack until we converged on the design